tests/node_test.cc

```diff
@@ -249,4 +250,51 @@ TEST(NodeTest, MultipleSubscriber) {
     }
 }
 
+TEST(NodeTest, StrandSubscriber) {
+    static constexpr std::size_t kThreadNum       = 4;
+    constexpr std::size_t        kNumOfSubChannel = 4;
+    constexpr std::size_t        kMessageNumber   = 50000;
+
+    using TestMessageType = TestMessage<11>;
+
+    CRNode publisher;
+    CRNode subscriber;
+
+    auto runner = JobRunner::MakeJobRunner(JobRunner::Config{
+        .thread_num_ = kThreadNum,
+    });
+    subscriber.SetRunner(runner);
+
+    std::mutex                                cv_mutex;
+    std::condition_variable                   cv;
+    std::array<std::size_t, kNumOfSubChannel> channel_counter{0};
+    std::array<bool, kNumOfSubChannel>        complete{0};
+
+    for (std::size_t channel_subid = 0; channel_subid < kNumOfSubChannel; ++channel_subid) {
+        subscriber.Subscribe<TestMessageType>(
+            channel_subid,
+            [&counter = channel_counter[channel_subid], &complete = complete[channel_subid], &cv_mutex, &cv](
+                const std::shared_ptr<TestMessageType>& message) {
+                EXPECT_EQ(counter++, message->value_);
+                if (counter == kMessageNumber) {
+                    std::unique_lock cv_lock(cv_mutex);
```

Comment:
mtx/lck?
Typically I abbrev those to distinguish from STL types.

Reply:
What I recommended are usually "standard" abbrev widely used. There're special systems of abbrev for 3-5 char, used by asm, 8.3, similar cmdline width limits in the early days, and alignment with them.
I would not use abbr for abbrev though in most cases though, as abbrev/abbrv are already commonly used, while abbr is similar to addr and doesn't carry enough consonants.

Here it's not for shortening, but just for systematic collision avoidance.

Reply:
Ack